supybot_fedora/plugin.py

```diff
@@ -168,17 +170,32 @@ def __init__(self, irc):
 
         # caches, automatically downloaded on __init__, manually refreshed on
         # .refresh
-        self.userlist = None
+        self.users = None
+        self.faslist = None
+        self.nickmap = None
 
         # To get the information, we need a username and password to FAS.
         # DO NOT COMMIT YOUR USERNAME AND PASSWORD TO THE PUBLIC REPOSITORY!
         self.fasurl = self.registryValue("fas.url")
         self.username = self.registryValue("fas.username")
         self.password = self.registryValue("fas.password")
 
-        self.fasclient = AccountSystem(
-            self.fasurl, username=self.username, password=self.password
-        )
+        # Use FASJSON
+        if self.registryValue("use_fasjson"):
+            try:
+                self.fasjsonclient = fasjson_client.Client(
+                    url=self.registryValue("fasjson.url"),
+                )
+            except fasjson_client.errors.ClientSetupError as e:
+                self.log.error(
+                    "Something went wrong setting up "
+                    "fasjson client with error: %s" % e
+                )
+                return {}
+        else:
+            self.fasclient = AccountSystem(
+                self.fasurl, username=self.username, password=self.password
+            )
```

Comment:
...maybe the `fedora.client.fas2` or `fasjson_client` imports should go here. Or be wrapped like this at the top of the module:

````
try:
    from fedora.client.fas2 import AccountSystem
except ImportError:
    AccountSystem = None

try:
    import fasjson_client
except ImportError:
    fasjson_client = None
````

...and then give diagnostic info if the configured system can't be imported.